anaconda_anon_usage/utils.py

```diff
@@ -48,13 +56,46 @@ def _random_token(what="random"):
     return result
 
 
-def _saved_token(fpath, what):
+def _final_attempt():
+    """
+    Called upon the graceful exit from conda, this attempts to
+    write an environment token that was deferred because the
+    environment directory was not yet available.
+    """
+    global DEFERRED
+    for must_exist, fpath, token in DEFERRED:
+        _write_attempt(must_exist, fpath, token)
+
+
+atexit.register(_final_attempt)
```

Comment:
BTW, conda also has a post-command plugin hook, that could be used for this final attempt, wondering if it might be good to use the `atexit` register only for the versions that don't support plugins?

Reply:
Agreed, that's a good idea